### PR TITLE
Optimizations for PUT message rate

### LINKED ISSUE #2815: No unsignaled WQEs for puts?
This is more of a question than an issue: from https://github.com/openucx/ucx/blob/master/src/uct/ib/rc/accel/rc_mlx5_common.h#L579-L614
I see that the WQEs are always signaled for puts but signaling is moderated for active messages (every 64 WQEs). What was the design motivation behind moderating signaling only for active messages not for puts?

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2815

### CODE REVIEW

**Comment by yosefe** [comment]
@shamisp @rzambre

File: src/ucp/rma/rma_basic.c [hunk 1]
```diff
@@ -24,7 +24,9 @@ static ucs_status_t ucp_rma_basic_progress_put(uct_pending_req_t *self)
     ucs_assert(rkey->cache.ep_cfg_index == ep->cfg_index);
     ucs_assert(rkey->cache.rma_lane == lane);
 
-    if (req->send.length <= ucp_ep_config(ep)->bcopy_thresh) {
+    if ((req->send.length <= rma_config->max_put_short) ||
```
[the remark this answers is not in the document]
Comment (yosefe) [reply src/ucp/rma/rma_basic.c]: we have an option to use fragmented put_short, for sizes in range max_put_short..bcopy_thresh
it is enabled if the user sets bcopy_thresh to some value
however default is 0, which (by mistake) disabled put_short completely and used put_bcopy instead

[the remark this answers is not in the document]
Comment (brminich) [reply src/ucp/rma/rma_basic.c]: ok, got it

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8215/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@rzambre FYI

**Comment by rzambre** [comment]
Thanks!

**Comment by shamisp** [comment]
@yosefe how flush is implemented in this case ?

**Comment by yosefe** [comment]
flush may have more overhead with rc_mlx5. but a larger-or-equal overhead is removed from PUT. so in total, it should be same or better.

**Pull request merged**